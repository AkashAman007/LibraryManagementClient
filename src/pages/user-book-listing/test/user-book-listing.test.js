import React from 'react';
import {render, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import  UserBookListing  from "../user-book-listing";
import { MemoryRouter } from 'react-router-dom';

afterEach(cleanup);

it(" renders user book listing", async () => {
    await wait()
    const tree = render(<MemoryRouter><UserBookListing/></MemoryRouter>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it("library books title", async () => {
    await wait();
    const tree = render(<MemoryRouter><UserBookListing/></MemoryRouter>)
    const titleElement = tree.getByText("My Books");
    expect(titleElement).toBeTruthy;
    expect(titleElement).toHaveClass('title');
});

it("user book listing has link to library books", async () => {
    await wait();
    const tree = render(<MemoryRouter><UserBookListing/></MemoryRouter>)
    const linkElement = tree.getByTestId("library-books-link");
    expect(linkElement).toBeTruthy;
    expect(linkElement).toHaveClass('link');
    expect(linkElement).toHaveAttribute('href');
    expect(linkElement.getAttribute('href')).toEqual('/');
});

it("when borrowed books list is empty", async () => {
    await wait();
    const tree = render(<MemoryRouter><UserBookListing/></MemoryRouter>)
    const messageElement = tree.getByText("You don't have any borrowed book");
    expect(messageElement).toBeTruthy;
});
