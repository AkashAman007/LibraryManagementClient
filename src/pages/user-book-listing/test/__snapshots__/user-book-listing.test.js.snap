// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders user book listing 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="title"
    >
      My Books
      <a
        class="float-right link"
        data-testid="library-books-link"
        href="/"
      >
        View Books
      </a>
    </h1>
    <div
      class="no-content"
    >
      You don't have any borrowed book
    </div>
  </div>
</DocumentFragment>
`;
